core-api/src/main/java/com/optimizely/ab/config/Experiment.java

```diff
@@ -173,6 +173,103 @@ public boolean isLaunched() {
         return status.equals(ExperimentStatus.LAUNCHED.toString());
     }
 
+    public String serializeConditions(Map<String, String> audiencesMap) {
+
+        Condition condition = this.audienceConditions;
+
+        return condition instanceof EmptyCondition ? "" : this.serialize(condition, audiencesMap);
+    }
+
+    private String getNameFromAudienceId(String audienceId, Map<String, String> audiencesMap) {
+        String audienceName = audiencesMap.get(audienceId);
+        return !audienceName.isEmpty() ? "\"" + audienceName + "\"" : "\"" + audienceId + "\"";
+    }
+
+    private String getOperandOrAudienceId(Condition condition, Map<String, String> audiencesMap){
+        String operand = "";
+        if (condition != null) {
+            if (condition instanceof AndCondition) {
+                operand = "AND";
+            } else if (condition instanceof NotCondition) {
+                operand = "NOT";
+            } else if (condition instanceof OrCondition) {
+                operand = "OR";
+            } else if (condition instanceof AudienceIdCondition) {
+                String audienceName = this.getNameFromAudienceId(((AudienceIdCondition<?>) condition).getAudienceId(),
+                    audiencesMap);
+                operand = audienceName;
+            }
+        }
+        return operand;
+    }
+
+    public String serialize(Condition condition, Map<String, String> audiencesMap) {
```

Comment:
This looks great. No need to touch Condition Interface :)

Reply:
Thank you!